### PR TITLE
DDP-5114-gae-service-to-log: detect GAE service name from env var GAE…

### PR DESCRIPTION
**Pull request by evgeniipr**
…_SERVICE.

## Context

Modified **HostUtil.java**: added method to get GAE service name which detected from env variable GAE_SERVICE. 
The service name added at the beginning of a message sent to Slack alert channel.
For example: [test-GAE-service]

Additional changes: Defined latest version of Mockito (in **parent-pom.xml**) using which the static methods mocking is possible; in unit tests do mocking of environment variables via **SystemUtil.java** methods mocking.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [x] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

**Review by yufengwng — CHANGES_REQUESTED** [review]
I have some concerns:

1. I don't understand why we need to do this. GCP Logging already tracks which GAE service a log entry comes from. For example, in the GCP Log Viewer you can query with `resource.labels.module_id="pepper-backend"` to find logs from the `pepper-backend` service. So I don't think we need to add yet another parameter to the log format.
2. If I understand the ticket, I believe what @MocanaAtBroad wants is to have the GAE service name show up in the _Slack alerts_. What this means to me is we should change `SlackAppender` to read the `GAE_SERVICE` env variable instead of fiddling with `HostUtil` or `MDC`.

**Comment by zyme** [comment]
I share @yufengwng's concerns.  I think I didn't write the ticket very well--my apologies!  The key thing is indeed to get the service passed through to slack.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/environment/HostUtil.java [hunk 1]
```diff
@@ -1,39 +1,100 @@
...
+     * Example of a substring added to log message: [GAE_INST_XX:GAE_SERV_XX]
+     */
+    public static void putGAEInstanceAndGAEServiceToMDC() {
+        MDC.put(GAE_INSTANCE_AND_SERVICE_LOG_PARAM, generateGAEInstanceAndGAEServiceForLogging());
+    }
+
+    /**
+     * Needed for testing purposes
+     */
+    @VisibleForTesting
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/environment/HostUtil.java:72]: Never seen this annotation before. Nice!

**Review by MocanaAtBroad — COMMENTED** [review]
Yeah. I think it will be easy for Evgenii to fix though. He figured out how to get the GCP metadata. Looks like just some tweaks to the SlackAppender ought to do trick.

**Comment by evgeniipr** [comment]
Hi!
So, it no need to modify HostUtil and just to read GAE_SERVICE directly in SlackAppender and put it at the beginning of the message?

**Review by zyme — APPROVED** [review]

**Review by yufengwng — APPROVED** [review]
Looks about right.

**Pull request merged**